### PR TITLE
Analytics 6 release 4 3

### PR DESCRIPTION
**Pull request by yspringfield**
Fixed: #8610
Fixed: #8705

### CODE REVIEW

File: apps/festival/festival/src/app/dashboard/analytics/buyer/buyer-analytics.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<header class="dark-contrast-theme main" bgAsset="background_header.png" bgTheme="dark">
+<header #header class="dark-contrast-theme main" bgAsset="background_header.png" bgTheme="dark">
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/analytics/buyer/buyer-analytics.component.html:1]: why do you need `#header` ?

File: apps/festival/festival/src/app/dashboard/analytics/buyer/buyer-analytics.component.ts [hunk 2]
```diff
@@ -197,6 +198,12 @@ export class BuyerAnalyticsComponent {
     @Inject(APP) public app: App
   ) { }
 
+  ngAfterViewInit(): void {
+    setTimeout(() => {
+      scrollIntoView(this.header.nativeElement);
+    })
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/analytics/buyer/buyer-analytics.component.ts:204]:
```suggestion
    });
```

What is default duration if you don't provide one ?

File: libs/analytics/src/lib/components/bar-chart/bar-chart.component.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { ChangeDetectionStrategy, Component, Input, ViewChild } from "@angular/core";
 import { AnalyticData } from "@blockframes/model";
+import { trimString } from "@blockframes/utils/pipes";
```
Comment (phpgeekfr) [inline libs/analytics/src/lib/components/bar-chart/bar-chart.component.ts:3]:
```suggestion
import { ChangeDetectionStrategy, Component, Input, ViewChild } from '@angular/core';
import { AnalyticData } from '@blockframes/model';
import { trimString } from '@blockframes/utils/pipes';
```

File: libs/utils/src/lib/browser/utils.ts [hunk 4]
```diff
@@ -19,9 +19,6 @@ export function isSafari() {
 export function isChrome() { return navigator.userAgent.indexOf("Chrome") != -1 }
 
 export function scrollIntoView(element: Element) {
-  if (isChrome())
-    element.scrollIntoView();
-  else
     element.scrollIntoView({ behavior: 'smooth' });
```
Comment (phpgeekfr) [inline libs/utils/src/lib/browser/utils.ts:22]: bad indentation

File: libs/utils/src/lib/browser/utils.ts [hunk 5]
```diff
@@ -19,9 +19,6 @@ export function isSafari() {
 export function isChrome() { return navigator.userAgent.indexOf("Chrome") != -1 }
 
 export function scrollIntoView(element: Element) {
```
Comment (phpgeekfr) [inline libs/utils/src/lib/browser/utils.ts:21]: `scrollIntoView` is used a lot. have you checked that removing isChrome is not an issue ?

File: libs/analytics/src/lib/components/bar-chart/bar-chart.component.ts [hunk 6]
```diff
@@ -70,13 +71,13 @@ export class BarChartComponent {
 
   @Input() set data(analytics: AnalyticData[]) {
     if (!analytics) return;
-    
+
     const data = Object.values(analytics).map(d => d.count);
     this.barChartOptions.series = data.length ? [{ data }] : [];
-    this.barChartOptions.labels = Object.values(analytics).map(d => d.label);
+    this.barChartOptions.labels = Object.values(analytics).map(d => trimString(d.label, 16, false));
```
Comment (phpgeekfr) [inline libs/analytics/src/lib/components/bar-chart/bar-chart.component.ts:77]: maybe update `trimString` to `export function trimString(string: string, length: number, keepLastWordComplete?: boolean) {`

so you can remove `false`

File: libs/utils/src/lib/browser/utils.ts (hunk 5 above)
Comment (yspringfield) [reply libs/utils/src/lib/browser/utils.ts:21]: It is supported on Chrome.
https://caniuse.com/scrollintoview
Bildschirmfoto 2022-06-22 um 7 48 19 AM

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**